### PR TITLE
[SPARK-20607][CORE]Add new unit tests to ShuffleSuite

### PR DESCRIPTION
**Pull request by heary-cao**
## What changes were proposed in this pull request?

This PR update to two:
1.adds the new unit tests.
  testing would be performed when there is no shuffle stage, 
  shuffle will not generate the data file and the index files.
2.Modify the '[SPARK-4085] rerun map stage if reduce stage cannot find its local shuffle file' unit test, 
  parallelize is 1 but not is 2, Check the index file and delete.

## How was this patch tested?
The new unit test.

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/ShuffleSuite.scala [hunk 1]
```diff
@@ -271,25 +271,49 @@ abstract class ShuffleSuite extends SparkFunSuite with Matchers with LocalSparkC
...
+      indexFile.delete()
+    }
 
     // This count should retry the execution of the previous stage and rerun shuffle.
     rdd.count()
   }
 
+  test("cannot find its local shuffle file if no execution of the stage and rerun shuffle") {
+    val myConf = conf.clone().set("spark.test.noStageRetry", "true")
+    sc = new SparkContext("local", "test", myConf)
```
Comment (srowen) [inline core/src/test/scala/org/apache/spark/ShuffleSuite.scala]: You aren't closing the existing sc here

File: core/src/test/scala/org/apache/spark/ShuffleSuite.scala [hunk 2]
```diff
@@ -271,25 +271,49 @@ abstract class ShuffleSuite extends SparkFunSuite with Matchers with LocalSparkC
...
+    val rdd = sc.parallelize(1 to 10, 1).map((_, 1)).reduceByKey(_ + _)
+
+    // Cannot find one of the local shuffle blocks.
+    val hashFile = sc.env.blockManager.diskBlockManager.getFile(new ShuffleBlockId(0, 0, 0))
+    val sortFile = sc.env.blockManager.diskBlockManager.getFile(new ShuffleDataBlockId(0, 0, 0))
+    val indexFile = sc.env.blockManager.diskBlockManager.getFile(new ShuffleIndexBlockId(0, 0, 0))
+    assert(!hashFile.exists() && !sortFile.exists() && !indexFile.exists())
+
+     rdd.count()
```
Comment (srowen) [inline core/src/test/scala/org/apache/spark/ShuffleSuite.scala]: Indent is off

File: core/src/test/scala/org/apache/spark/ShuffleSuite.scala [hunk 3]
```diff
@@ -271,25 +271,49 @@ abstract class ShuffleSuite extends SparkFunSuite with Matchers with LocalSparkC
   test("[SPARK-4085] rerun map stage if reduce stage cannot find its local shuffle file") {
     val myConf = conf.clone().set("spark.test.noStageRetry", "false")
     sc = new SparkContext("local", "test", myConf)
-    val rdd = sc.parallelize(1 to 10, 2).map((_, 1)).reduceByKey(_ + _)
+    val rdd = sc.parallelize(1 to 10, 1).map((_, 1)).reduceByKey(_ + _)
```
Comment (srowen) [inline core/src/test/scala/org/apache/spark/ShuffleSuite.scala]: Why this change?

**Review by srowen — CHANGES_REQUESTED** [review]

File: core/src/test/scala/org/apache/spark/ShuffleSuite.scala (hunk 3 above)
Comment (heary-cao) [inline core/src/test/scala/org/apache/spark/ShuffleSuite.scala]: @srowen 
I think it over access to shuffle the data file with the index file closer to the test target. so modify it.

File: core/src/test/scala/org/apache/spark/ShuffleSuite.scala (hunk 2 above)
Comment (heary-cao) [inline core/src/test/scala/org/apache/spark/ShuffleSuite.scala]: sorry,
I have modify it.
thanks.

File: core/src/test/scala/org/apache/spark/ShuffleSuite.scala (hunk 3 above)
Comment (heary-cao) [inline core/src/test/scala/org/apache/spark/ShuffleSuite.scala]: Sorry,
When debugging use value, forget the back.

File: core/src/test/scala/org/apache/spark/ShuffleSuite.scala (hunk 1 above)
Comment (heary-cao) [inline core/src/test/scala/org/apache/spark/ShuffleSuite.scala]: Has been modified.
please review it again.
thanks.

**Comment by heary-cao** [comment]
@srowen 
I have modify this` Scala style`.
reset please.
thanks.

**Comment by srowen** [comment]
merged to master
